emily/README.md

```diff
@@ -0,0 +1,138 @@
+```
+
+This file is referenced within the cdk and deployed to the lambda.
+
+### Lambda Configuration
+
+The emily lambda takes in config values via the environment it's deployed on. As of writing this there are 6 environment values that are passed to emily during deployment. This can be found within the cdk definition.
+
+```javascript
+environment: {
+    DEPOSIT_TABLE_NAME: depositTableName,
+    WITHDRAWAL_TABLE_NAME: withdrawalTableName,
+    CHAINSTATE_TABLE_NAME: chainstateTableName,
+    LIMIT_TABLE_NAME: limitTableName,
+    IS_LOCAL: "true" | "false",
+    TRUSTED_REORG_API_KEY: trustedReorgApiKey,
+},
+```
+
+All these parameters are consumed via the `context.rs` file within the handler crate.
+
+The table names specify the dynamodb tables that the API uses to store data, and the trusted reorg api key is the api key that has the special ability to intiate a reorg in the chainstate. More on that later.
+
+## CDK
+
+The Emily Typescript CDK deploys a number of resources:
+
+1. Every DynamoDB table used by Emily
+2. The Emily rust lambda handler
+3. The API gateway instance that connects to the lambda and all connected API keys
+
+### Template Configuration
+
+As of writing this the below list of environment variables is the exhaustive set of configuration options of the cdk template. All of these and the nuances of their inclusion can be found within `cdk/lib/emily-stack-util.ts`.
+
+```bash
+# The stage to deploy to
+AWS_STAGE=dev
+
+# The AWS region on which to deploy the stack
+AWS_REGION=us-west-2
+
+# The number of API keys to generate
+NUM_SIGNER_API_KEYS=15
+
+# Your domain name.
+CUSTOM_ROOT_DOMAIN_NAME=<your-url.com>
+
+# the hosted zone id of the domain name, taken from AWS console after
+# purchasing your domain.
+HOSTED_ZONE_ID=Z00000000000000000000
+
+# The api key that you trust to indicate valid reorgs on the
+# api chainstate.
+TRUSTED_REORG_API_KEY=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+
+# Whether to deploy only the dynamodb tables - helpful for testing.
+TABLES_ONLY=false
+```
+
+> **Caution:** when deploying to a stage that's deemed to be for an active development stack (`dev`, `local`, `unit-test`) all persistent resources will be set to be deleted when the stack is redeployed or deleted. It's not recommended to use any of these environments for anything you want to be remotely persistent.
+
+### Custom Domains
+
+When you specify a custom domain name for your stack you need to also provide the hosted zone id which you can find in AWS console. You need to have purchased this before attempting to deploy the API to a custom domain. If you don't provide a custom root domain name it won't bother attaching the api to a custom domain, so you're safe if you don't want to purchase one.
+
+The custom domain will generate a different api based on the stage that you deploy to. Lets assume your custom domain is `example.com`
+
+- stage = `prod` -> `example.com`
+- stage = `dev` -> `dev.example.com`
```

Comment:
don't we also append the stage name? i.e. currently dev is hosted on dev.sbtc-emily-dev.com

Reply:
No, that's just because that's the url I bought, but that was fully manual.

Reply:
Mind if I resolve this?